src/libmongoc/src/mongoc/mongoc-stream-socket.c

```diff
@@ -36,10 +36,10 @@ struct _mongoc_stream_socket_t {
 static BSON_INLINE int64_t
 get_expiration(int32_t timeout_msec)
 {
-   if (timeout_msec < 0) {
-      return -1;
-   } else if (timeout_msec == 0) {
+   if (timeout_msec == MONGOC_SOCKET_TIMEOUT_IMMEDIATE) {
       return 0;
+   } else if (timeout_msec <= 0) {
+      return -1;
```

Comment:
```suggestion
      return -1; // Infinite.
```
Suggest drive-by improvement documenting the meaning of `-1` here.